File: p_012.js
function* fibonacci() { // a generator function
  let counter = 0;
  let tringular = counter
  while (true){
    counter += 1
    tringular += counter
    yield tringular
  }
}

function div_counter(n){
  let divs = 1
  for (let j=1; j<=Math.floor(n/2) ;j++){
    if (n % j == 0){
      divs += 1
    }
  }
  return divs
}

for (let n of fibonacci()) {
  
  // truncate the sequence at 1000
  if (div_counter(n) >= 500) {
    console.log(n);
    break;
  }
}

File: p_012.py
import math, time

def tringular_gen():
	counter = 0
	tringular = counter
	while True:		
		counter+=1
		tringular += counter
		yield tringular
		

def divisors(n):
	div_ = 1
	for i in range(1, math.ceil(n**0.5)):
		if n % i == 0:
			div_ += 1
	return div_


for i in tringular_gen():
	# print(i)
	div = divisors(i)
	if  div > 500:
		print(div, " got it " , i)
		break
	# time.sleep(2)